sql/plan/drop_view.go

```diff
@@ -0,0 +1,148 @@
+package plan
+
+import (
+	"github.com/src-d/go-mysql-server/sql"
+	errors "gopkg.in/src-d/go-errors.v1"
+)
+
+var errDropViewChild = errors.NewKind("any child of DropView must be of type SingleDropView")
+
+type SingleDropView struct {
+	database sql.Database
+	viewName string
+}
+
+// NewSingleDropView creates a SingleDropView.
+func NewSingleDropView(
+	database sql.Database,
+	viewName string,
+) *SingleDropView {
+	return &SingleDropView{database, viewName}
+}
+
+// Children implements the Node interface. It always returns nil.
+func (dv *SingleDropView) Children() []sql.Node {
+	return nil
+}
+
+// Resolved implements the Node interface. This node is resolved if and only if
+// its database is resolved.
+func (dv *SingleDropView) Resolved() bool {
+	_, ok := dv.database.(sql.UnresolvedDatabase)
+	return !ok
+}
+
+// RowIter implements the Node interface. It always returns an empty iterator.
+func (dv *SingleDropView) RowIter(ctx *sql.Context) (sql.RowIter, error) {
+	return sql.RowsToRowIter(), nil
+}
+
+// Schema implements the Node interface. It always returns nil.
+func (dv *SingleDropView) Schema() sql.Schema { return nil }
+
+// String implements the fmt.Stringer interface, using sql.TreePrinter to
+// generate the string.
+func (dv *SingleDropView) String() string {
+	pr := sql.NewTreePrinter()
+	_ = pr.WriteNode("SingleDropView(%s.%s)", dv.database.Name(), dv.viewName)
+
+	return pr.String()
+}
+
+// WithChildren implements the Node interface. It only succeeds if the length
+// of the specified children equals 0.
+func (dv *SingleDropView) WithChildren(children ...sql.Node) (sql.Node, error) {
+	if len(children) != 0 {
+		return nil, sql.ErrInvalidChildrenNumber.New(dv, len(children), 0)
+	}
+
+	return dv, nil
+}
+
+// Database implements the Databaser interfacee. It returns the node's database.
+func (dv *SingleDropView) Database() sql.Database {
+	return dv.database
+}
+
+// Database implements the Databaser interface, and it returns a copy of this
+// node with the specified database.
+func (dv *SingleDropView) WithDatabase(database sql.Database) (sql.Node, error) {
+	newDrop := *dv
+	newDrop.database = database
+	return &newDrop, nil
+}
+
+// DropView is a node representing the removal of a list of views, defined by
+// the children member. The flag ifExists represents whether the user wants the
+// node to fail if any of the views in children does not exist.
+type DropView struct {
+	children []sql.Node
+	Catalog  *sql.Catalog
+	ifExists bool
+}
+
+// NewDropView creates a DropView node with the specified parameters,
+// setting its catalog to nil.
+func NewDropView(children []sql.Node, ifExists bool) *DropView {
```

Comment:
Instead of having SingleDropView and DropView, what do you think about just keeping DropView and a single drop view would just be a DropView with a single children?

Reply:
I don't know if I'm following you :thinking: The `SingleDropView` is just a dummy node containing the view to drop, and I need to have it as one entity so every droppable view can contain its own database to resolve. But then I need them all together in a single node so I can call `DeleteList` and make sure that I check the errors and perform the actual removal in one atomic operation.

Reply:
Oh, nevermind then, I did not understood correctly what it was doing at first glance

Reply:
@erizocosmico maybe the names are not the best ones, I understand the confusion.